mojo/stdlib/stdlib/collections/result.mojo

```diff
@@ -0,0 +1,713 @@
+
+        Returns:
+            A reference to the contained data of the `Result` as a reference.
+
+        Notes:
+            This will abort on `Result` with an error.
+        """
+
+        if not self.__bool__():
+            abort(
+                "`Result.value()` called on `Result` with an error."
+                " Consider using `if Result:` to check whether the `Result`"
+                " is empty before calling `.value()`, or use `.or_else()`"
+                " to provide a default value."
+            )
+
+        return self.unsafe_value()
+
+    fn err_value(ref self) -> ref [self._value] Self.E:
+        """Get the error in the Result or a defaulted version of it.
+
+        Returns:
+            The error in the Result or a defaulted version of it.
+
+        Notes:
+            This will abort on `Result` without an error.
+        """
+        if self.__bool__():
+            abort(
+                "`Result.err_value()` called on `Result` without an error."
+                " Consider using `if Result:` to check whether the `Result`"
+                " is empty before calling `.value()` to provide a default "
+                "value."
+            )
+        return self.err_unsafe_value()
+
+    @always_inline
+    fn unsafe_value(ref self) -> ref [self._value] Self.T:
+        """Unsafely retrieve a reference to the value of the `Result`.
+
+        Returns:
+            A reference to the contained data of the `Result` as a reference.
+
+        Notes:
+            This will **not** abort on `Result` with an error.
+        """
+
+        debug_assert(self.__bool__(), "`.value()` on `Result` with an error")
+        return self._value.unsafe_get[Self._container_type]()[0]
+
+    fn err_unsafe_value(ref self) -> ref [self._value] Self.E:
+        """Get the error in the Result or a defaulted version of it.
+
+        Returns:
+            The error in the Result or a defaulted version of it.
+
+        Notes:
+            This will **not** abort on `Result` without an error.
+        """
+        debug_assert(
+            self.__bool__(), "`.err_value()` on `Result` without an error"
+        )
+        return self._value.unsafe_get[Self.E]()
+
+    fn take(mut self) -> Self.T:
+        """Move the value out of the `Result`.
+
+        Returns:
+            The contained data of the `Result` as an owned T value.
+
+        Notes:
+            This will abort when trying to take a value from a `Result`
+            with an error.
+        """
+        if not self.__bool__():
+            abort(
+                "`Result.take()` called trying to take a value from a"
+                "`Result` with an error. Consider"
+                " using `if Result:` to check whether the `Result` is empty"
+                " before calling `.take()`, or use `.or_else()` to provide"
+                " a default value."
+            )
+        return self.unsafe_take()
+
+    fn err_take(mut self) -> Self.E:
```

Comment:
Having this be name `take_err()` reads more cleanly than `err_take`. It should also provide parity by having a `take_` prefix.

Ditto with `unsafe_` versions.

IMO: 
`take()`
`take_err()`
`unsafe_take()`
`unsafe_take_err()`